maingame.py

```diff
@@ -0,0 +1,688 @@
+                    is_selectable = False
+                    print('MSG: You can only select units that are in the same tile together!')
+                    break
+
+        if(is_selectable == True) and (unit.owner == player):
+            is_selected = unit.toggle_select()
+            if (is_selected == 1):
+                selected.append(unit)
+            else:
+                selected.remove(unit)
+                unit_exists = False
+                for element in selected:
+                    if type(element) is Unit:
+                        unit_exists = True
+                        break
+                if unit_exists == False:
+                    for tile in selected:
+                        tile.toggle_select()
+                        selected.remove(tile)
+        elif (unit.owner != player):
+            print('MSG: You can only select your own units!')
+
+
+class View(object):
+    '''DOCSTRING
+        DESC: view editor
+        ATTR:
+        self - unit controller object - self-referential
+        model - object - thing that is being controlled
+        MTDS:
+        __init__ - initializes view controll object
+        draw - blits unit onto surface
+        '''
+    def __init__(self, model):
+        '''DOCSTRING
+            DESC: init
+            ARGS:
+            self - View object, self-referential
+            model - object to be shown
+            RTRN: none
+            SHOW: none
+            '''
+        self.model = model
+
+    def draw(self, surface):
+        '''DOCSTRING
+            DESC: update what the screen should look like
+                Does not redraw screen, just updates desired screen
+            ARGS:
+            self - View object, self-referential
+            surface - object to be shown
+            RTRN: none
+            SHOW: none
+            '''
+        model = self.model
+        surface.blit(self.model.icon, (self.model.x, self.model.y))
+
+
+class MapGenerator(object):
+    '''DOCSTRING
+        DESC: Class to generate board
+        ATTR:
+        self - unit controller object - self-referential
+        MTDS:
+        __init__ - initializes the mapgenerator (not the map!)
+        generate - generates a map including two bases
+        adjacency_matrix_gen - generates a matrix of adjacencies between tiles for generated map
+        check_adjacent - checks if two tiles are adjacent
+        '''
+    def __init__(self, screensize, xtiles=8, ytiles=6):
+        self.x = screensize[0]
+        self.y = screensize[1]
+        self.xtiles = xtiles
+        self.ytiles = ytiles
+        self.tiles = []
+        self.tileviews = []
+
+    def generate(self):
+        # Randomly generates tiles for bases
+        randx_1 = random.randint(0,(self.xtiles-1)//4)
+        randx_2 = random.randint(3*self.xtiles//4,self.xtiles-1)
+        randy_1 = random.randint(0,self.ytiles-1)
+        randy_2 = random.randint(0,self.ytiles-1)
+        # num_tiles = self.xtiles*self.ytiles
```

Comment:
Delete extraneous lines that you used during development but is not a part of your final code.